packages/google_mlkit_commons/android/src/main/java/com/google_mlkit_commons/InputImageConverter.java

```diff
@@ -21,27 +22,39 @@ public static InputImage getInputImageFromData(Map<String, Object> imageData,
         //Differentiates whether the image data is a path for a image file or contains image data in form of bytes
         String model = (String) imageData.get("type");
         InputImage inputImage;
-        if (model.equals("file")) {
+        if (model != null && model.equals("file")) {
             try {
                 inputImage = InputImage.fromFilePath(context, Uri.fromFile(new File(((String) imageData.get("path")))));
                 return inputImage;
             } catch (IOException e) {
                 Log.e("ImageError", "Getting Image failed");
-                e.printStackTrace();
+                Log.e("ImageError", e.toString());
                 result.error("InputImageConverterError", e.toString(), null);
                 return null;
             }
-        } else if (model.equals("bytes")) {
-            Map<String, Object> metaData = (Map<String, Object>) imageData.get("metadata");
-            inputImage = InputImage.fromByteArray((byte[]) imageData.get("bytes"),
-                    (int) (double) metaData.get("width"),
-                    (int) (double) metaData.get("height"),
-                    (int) metaData.get("rotation"),
-                    (int) metaData.get("image_format"));
-            return inputImage;
         } else {
-            result.error("InputImageConverterError", "Invalid Input Image", null);
-            return null;
+            if (model != null && model.equals("bytes")) {
+                try {
+                    @SuppressWarnings("unchecked")
+                    Map<String, Object> metaData = (Map<String, Object>) imageData.get("metadata");
+
+                    assert metaData != null;
+                    inputImage = InputImage.fromByteArray((byte[]) Objects.requireNonNull(imageData.get("bytes")),
+                            Double.valueOf((String) metaData.get("width")).intValue(),
+                            Double.valueOf((String) metaData.get("height")).intValue(),
+                            Integer.parseInt((String) metaData.get("rotation")),
+                            InputImage.IMAGE_FORMAT_NV21);
```

Comment:
you are hardcoding the format to `InputImage.IMAGE_FORMAT_NV21`, use `metaData.get("image_format")` instead

Reply:
Sure, I replaced this hardcoding and used `metaData.get("image_format")` instead

Reply:
@fbernaly At version 0.4.0, I see this 
![image](https://github.com/flutter-ml/google_ml_kit_flutter/assets/44032235/0825805a-c1cc-47b3-b7e2-d71de2e195d0)
Will replacing hardcoding have any effect?

Reply:
In version `0.7.2` we remove the hardcoding :
https://github.com/flutter-ml/google_ml_kit_flutter/pull/634/files